src/models/real-fevr-marketplace.ts

```diff
@@ -0,0 +1,128 @@
+import {Model} from '@base/model';
+import {Web3Connection} from '@base/web3-connection';
+import {Web3ConnectionOptions} from '@interfaces/web3-connection-options';
+import {Deployable} from '@interfaces/deployable';
+import RealFevrMarketplaceJson from '@abi/RealFevrMarketplace.json';
+import {RealFevrMarketplaceMethods} from '@methods/real-fevr-marketplace';
+import {AbiItem} from 'web3-utils';
+import {Errors} from '@interfaces/error-enum';
+import {ERC20} from '@models/erc20';
+import {RealFevrOpener} from '@models/real-fevr-opener';
+import {toSmartContractDecimals} from '@utils/numbers';
+
+export class RealFevrMarketplace extends Model<RealFevrMarketplaceMethods> implements Deployable {
+  constructor(
+    web3Connection: Web3Connection|Web3ConnectionOptions,
+    contractAddress?: string,
+    readonly collectiblesAddress?: string,
+    readonly tokenAddress?: string,
+  ) {
+    super(web3Connection, RealFevrMarketplaceJson.abi as AbiItem[], contractAddress);
+  }
+
+  private _isETHTransaction!: boolean;
+  isETHTransaction() { return this._isETHTransaction; }
+
+  private _erc20!: ERC20;
+  get erc20() { return this._erc20; }
+
+  private _opener!: RealFevrOpener;
+  get opener() { return this._opener; }
+
+  async loadContract() {
+    if (!this.contract)
+      await super.loadContract();
+
+    const tokenAddress = await this.getERC20TokenAddress() || this.tokenAddress;
+    if (!tokenAddress)
+      throw new Error(Errors.MissingERC20AddressOnContract);
+
+    const collectiblesAddress = await this.getERC721TokenAddress() || this.collectiblesAddress;
+    if (!collectiblesAddress)
+      throw new Error(Errors.MissingERC721AddressOnContract);
+
+    this._isETHTransaction = tokenAddress === '0x0000000000000000000000000000000000000000';
+
+    if (!this.isETHTransaction()) {
+      // Set Token Address Contract for easy access
+      this._erc20 = new ERC20(this.web3Connection, tokenAddress);
+      await this._erc20.loadContract();
+    }
+
+    this._opener = new RealFevrOpener(this.web3Connection, collectiblesAddress);
+    await this._opener.loadContract();
+  }
+
+  async start() {
+    await super.start();
+    await this.loadContract();
+  }
+
+  async deployJsonAbi(collectiblesAddress: string, tokenAddress: string = '0x0000000000000000000000000000000000000000') {
```

Comment:
don't provide mock values, if needed -- add a comment to the function head describing the arguments supposed value

Reply:
Wasn't really a mock value, the marketplace and the opener contracts can actually be initialized with the 0x0 address, at which point it will use the native coin of the chain rather than a provided token from a contract. I've made the argument mandatory and added a comment to account for this case.